services/workflows-service/src/workflow/workflow.service.ts

```diff
@@ -645,18 +663,69 @@ export class WorkflowService {
     const updatedWorkflow = await this.updateContextById(workflowId, {
       documents: documentsWithDecision,
     });
-
     const correlationId = await this.getCorrelationIdFromWorkflow(updatedWorkflow);
+    const entityId = updatedWorkflow.businessId || updatedWorkflow.endUserId;
 
     this.workflowEventEmitter.emit('workflow.context.changed', {
-      oldRuntimeData: runtimeData,
+      oldRuntimeData: workflow,
       updatedRuntimeData: updatedWorkflow,
       state: updatedWorkflow.state,
-      entityId: (updatedWorkflow.businessId || updatedWorkflow.endUserId) as string,
+      entityId,
       correlationId: correlationId,
     });
 
-    return updatedWorkflow;
+    // TODO: Check against `contextSchema` or a policy if the length of documents is equal to the number of tasks defined.
```

Comment:
Updates the workflow to completed if all documents have a decision and the workflow config is set with `completedWhenTasksResolved: true`